lib/AST/ASTMangler.cpp

```diff
@@ -3116,6 +3124,13 @@ void ASTMangler::appendAnyGenericType(const GenericTypeDecl *decl,
   // Always use Clang names for imported Clang declarations, unless they don't
   // have one.
   auto tryAppendClangName = [this, decl]() -> bool {
+    // We use the Swift name rather than the Clang name for compatibility
+    // aliases to avoid multiple declarations having the same mangled name.
```

Comment:
More precisely it's because `ClangImporterContext` entities are always mangled according to their Swift name, since that's what the demangler expects